### PR TITLE
Master test

### PR DESCRIPTION
**Pull request by massandharsh**

### CODE REVIEW

**Comment by massandharsh** [comment]
---

### 📝 Overall PR Feedback

Thank you for your contribution! The MR introduces a new DbCommandInfoSage class which seems to handle database file information retrieval. However, there are a few areas that could be improved:

1. Code quality: Some comments and print statements are unprofessional and should be removed or improved.
2. Magic numbers: Consider extracting magic numbers to named constants for better readability and maintainability.
3. Exception handling: Improve the exception handling by using a proper logging mechanism instead of printing to System.out.
4. Documentation: The new class could benefit from some high-level documentation explaining its purpose and how it fits into the overall system.

Please address these points, and feel free to ask if you need any clarification or assistance.

---

### ✅ Test Results Summary

The CI pipeline has passed, which is great! However, it would be beneficial to add unit tests for the new DbCommandInfoSage class to ensure its functionality and improve code coverage. Consider writing tests for both the verifyCommand and invoke methods, covering different scenarios including edge cases.

File: src/main/java/commands/CountFromTableCommand.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class CountFromTableCommand implements Commands<String>{
     private final int anInt = 2265232;
        private final int temp4 = 3;
     //This code is hard :(
-    //Can you explain it hard it is
+    //Can you explain it hard it is very hard
```
Comment (massandharsh) [inline src/main/java/commands/CountFromTableCommand.java:23]: 💬 **Category:** _Comment Quality_

Consider removing or improving this comment. It doesn't provide meaningful information about the code and may be considered unprofessional. 📝 **Tip:** Replace with a descriptive comment explaining the purpose or complexity of the upcoming code.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0NvdW50RnJvbVRhYmxlQ29tbWFuZC5qYXZhIiwibGluZSI6MjMsInNpZGUiOiJSSUdIVCIsImNhdGVnb3J5IjoiQ29tbWVudCBRdWFsaXR5IiwidHlwZSI6IlVQIiwiY29tbWVudCI6IkNvbnNpZGVyIHJlbW92aW5nIG9yIGltcHJvdmluZyB0aGlzIGNvbW1lbnQuIEl0IGRvZXNuJ3QgcHJvdmlkZSBtZWFuaW5nZnVsIGluZm9ybWF0aW9uIGFib3V0IHRoZSBjb2RlIGFuZCBtYXkgYmUgY29uc2lkZXJlZCB1bnByb2Zlc3Npb25hbC4gXHVEODNEXHVEQ0REICoqVGlwOioqIFJlcGxhY2Ugd2l0aCBhIGRlc2NyaXB0aXZlIGNvbW1lbnQgZXhwbGFpbmluZyB0aGUgcHVycG9zZSBvciBjb21wbGV4aXR5IG9mIHRoZSB1cGNvbWluZyBjb2RlLiIsImV4cCI6MTc1MDgzNjc2NX0.XgMv_AjU6AOIsXJdH79KOIMsIoXLyAyzW65Y4QEMjOw)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0NvdW50RnJvbVRhYmxlQ29tbWFuZC5qYXZhIiwibGluZSI6MjMsInNpZGUiOiJSSUdIVCIsImNhdGVnb3J5IjoiQ29tbWVudCBRdWFsaXR5IiwidHlwZSI6IkRPV04iLCJjb21tZW50IjoiQ29uc2lkZXIgcmVtb3Zpbmcgb3IgaW1wcm92aW5nIHRoaXMgY29tbWVudC4gSXQgZG9lc24ndCBwcm92aWRlIG1lYW5pbmdmdWwgaW5mb3JtYXRpb24gYWJvdXQgdGhlIGNvZGUgYW5kIG1heSBiZSBjb25zaWRlcmVkIHVucHJvZmVzc2lvbmFsLiBcdUQ4M0RcdURDREQgKipUaXA6KiogUmVwbGFjZSB3aXRoIGEgZGVzY3JpcHRpdmUgY29tbWVudCBleHBsYWluaW5nIHRoZSBwdXJwb3NlIG9yIGNvbXBsZXhpdHkgb2YgdGhlIHVwY29taW5nIGNvZGUuIiwiZXhwIjoxNzUwODM2NzY1fQ.xD_VuvTdX1dq1pLI02LG2IMajEeLahTBTlQgmPJ__g4)

File: src/main/java/commands/DbCommandInfoSage.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        String [] commands = commandExtractor(command);
+        return commands.length == 2 && getCommandEnum(commands[1]) == CommandEnum.DB_INFO;
+    }
+
+    @Override
+    public String invoke(String command) throws Exception {
+        if(!verifyCommand(command)){
+            throw new RuntimeException("Something went wrong with executed command");
+        }
+        System.out.println("Coding causes cancer");
```
Comment (massandharsh) [inline src/main/java/commands/DbCommandInfoSage.java:21]: 🧹 **Category:** _Code Smell_

This print statement seems out of place and unprofessional. Consider removing it or replacing it with meaningful logging. 🧹 **Tip:** Use a proper logging framework like SLF4J for better debugging and traceability.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0RiQ29tbWFuZEluZm9TYWdlLmphdmEiLCJsaW5lIjoyMSwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJDb2RlIFNtZWxsIiwidHlwZSI6IlVQIiwiY29tbWVudCI6IlRoaXMgcHJpbnQgc3RhdGVtZW50IHNlZW1zIG91dCBvZiBwbGFjZSBhbmQgdW5wcm9mZXNzaW9uYWwuIENvbnNpZGVyIHJlbW92aW5nIGl0IG9yIHJlcGxhY2luZyBpdCB3aXRoIG1lYW5pbmdmdWwgbG9nZ2luZy4gXHVEODNFXHVEREY5ICoqVGlwOioqIFVzZSBhIHByb3BlciBsb2dnaW5nIGZyYW1ld29yayBsaWtlIFNMRjRKIGZvciBiZXR0ZXIgZGVidWdnaW5nIGFuZCB0cmFjZWFiaWxpdHkuIiwiZXhwIjoxNzUwODM2NzY2fQ.t9FwlHRccF2ZDaaB67D6_x3sprB_EDKWdvAdMxGYl9Y)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0RiQ29tbWFuZEluZm9TYWdlLmphdmEiLCJsaW5lIjoyMSwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJDb2RlIFNtZWxsIiwidHlwZSI6IkRPV04iLCJjb21tZW50IjoiVGhpcyBwcmludCBzdGF0ZW1lbnQgc2VlbXMgb3V0IG9mIHBsYWNlIGFuZCB1bnByb2Zlc3Npb25hbC4gQ29uc2lkZXIgcmVtb3ZpbmcgaXQgb3IgcmVwbGFjaW5nIGl0IHdpdGggbWVhbmluZ2Z1bCBsb2dnaW5nLiBcdUQ4M0VcdURERjkgKipUaXA6KiogVXNlIGEgcHJvcGVyIGxvZ2dpbmcgZnJhbWV3b3JrIGxpa2UgU0xGNEogZm9yIGJldHRlciBkZWJ1Z2dpbmcgYW5kIHRyYWNlYWJpbGl0eS4iLCJleHAiOjE3NTA4MzY3NjZ9.Bxp-9WO_WQ1pfCBdWNuCNT5HB50bsJHkIjIp92ql-gA)

File: src/main/java/commands/DbCommandInfoSage.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        String [] commands = commandExtractor(command);
+        try(FileInputStream fis = new FileInputStream(commands[0])){
+            //First we have to skip the 16 bytes of this file as that is header
+            //SQLITE format 3 + null terminator
```
Comment (massandharsh) [inline src/main/java/commands/DbCommandInfoSage.java:25]: 📖 **Category:** _Readability_

Consider using a constant for the magic number 16. This improves readability and maintainability. 📖 **Tip:** Define a private static final int for this value at the class level.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0RiQ29tbWFuZEluZm9TYWdlLmphdmEiLCJsaW5lIjoyNSwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJSZWFkYWJpbGl0eSIsInR5cGUiOiJVUCIsImNvbW1lbnQiOiJDb25zaWRlciB1c2luZyBhIGNvbnN0YW50IGZvciB0aGUgbWFnaWMgbnVtYmVyIDE2LiBUaGlzIGltcHJvdmVzIHJlYWRhYmlsaXR5IGFuZCBtYWludGFpbmFiaWxpdHkuIFx1RDgzRFx1RENENiAqKlRpcDoqKiBEZWZpbmUgYSBwcml2YXRlIHN0YXRpYyBmaW5hbCBpbnQgZm9yIHRoaXMgdmFsdWUgYXQgdGhlIGNsYXNzIGxldmVsLiIsImV4cCI6MTc1MDgzNjc2N30.Y0QvU2AIeBu5TVyFqeJTGbBUXn2BtKp7Jwkw7y1B1vc)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0RiQ29tbWFuZEluZm9TYWdlLmphdmEiLCJsaW5lIjoyNSwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJSZWFkYWJpbGl0eSIsInR5cGUiOiJET1dOIiwiY29tbWVudCI6IkNvbnNpZGVyIHVzaW5nIGEgY29uc3RhbnQgZm9yIHRoZSBtYWdpYyBudW1iZXIgMTYuIFRoaXMgaW1wcm92ZXMgcmVhZGFiaWxpdHkgYW5kIG1haW50YWluYWJpbGl0eS4gXHVEODNEXHVEQ0Q2ICoqVGlwOioqIERlZmluZSBhIHByaXZhdGUgc3RhdGljIGZpbmFsIGludCBmb3IgdGhpcyB2YWx1ZSBhdCB0aGUgY2xhc3MgbGV2ZWwuIiwiZXhwIjoxNzUwODM2NzY3fQ.L7NuQ1mNq01twcn6iQ0ndU0ob1xiKsHmT-U8LFG4q-g)

File: src/main/java/commands/DbCommandInfoSage.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            fis.skip(16);
+            byte [] pageSizeBuffer = new byte[2]; //Big Endian value from left to right
+            fis.read(pageSizeBuffer);
+            //Now we convert to integer
+            short pagesShort = ByteBuffer.wrap(pageSizeBuffer).getShort();
+            int pageSize = Short.toUnsignedInt(pagesShort);
```
Comment (massandharsh) [inline src/main/java/commands/DbCommandInfoSage.java:31]: 📖 **Category:** _Readability_

The calculation for noOfBytesForCellReading uses magic numbers. Consider extracting these to named constants for better clarity. 📖 **Tip:** Create constants for 103 and 16 with descriptive names to explain their significance.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0RiQ29tbWFuZEluZm9TYWdlLmphdmEiLCJsaW5lIjozMSwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJSZWFkYWJpbGl0eSIsInR5cGUiOiJVUCIsImNvbW1lbnQiOiJUaGUgY2FsY3VsYXRpb24gZm9yIG5vT2ZCeXRlc0ZvckNlbGxSZWFkaW5nIHVzZXMgbWFnaWMgbnVtYmVycy4gQ29uc2lkZXIgZXh0cmFjdGluZyB0aGVzZSB0byBuYW1lZCBjb25zdGFudHMgZm9yIGJldHRlciBjbGFyaXR5LiBcdUQ4M0RcdURDRDYgKipUaXA6KiogQ3JlYXRlIGNvbnN0YW50cyBmb3IgMTAzIGFuZCAxNiB3aXRoIGRlc2NyaXB0aXZlIG5hbWVzIHRvIGV4cGxhaW4gdGhlaXIgc2lnbmlmaWNhbmNlLiIsImV4cCI6MTc1MDgzNjc2OX0.zT8X-d0n5d-pwfmr21KYTloBd5IAc68w9wo5kOZjDzM)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6NTksInJlcG8iOiJzcWwtbGl0ZSIsIm93bmVyIjoibWFzc2FuZGhhcnNoIiwiY29tbWl0X2lkIjoiNjhkNWI0ODdhY2E1OWM2OGRlMTI4ZGVkNTcwMjgzMGRkMzQ5ZWVhMCIsInBhdGgiOiJzcmMvbWFpbi9qYXZhL2NvbW1hbmRzL0RiQ29tbWFuZEluZm9TYWdlLmphdmEiLCJsaW5lIjozMSwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJSZWFkYWJpbGl0eSIsInR5cGUiOiJET1dOIiwiY29tbWVudCI6IlRoZSBjYWxjdWxhdGlvbiBmb3Igbm9PZkJ5dGVzRm9yQ2VsbFJlYWRpbmcgdXNlcyBtYWdpYyBudW1iZXJzLiBDb25zaWRlciBleHRyYWN0aW5nIHRoZXNlIHRvIG5hbWVkIGNvbnN0YW50cyBmb3IgYmV0dGVyIGNsYXJpdHkuIFx1RDgzRFx1RENENiAqKlRpcDoqKiBDcmVhdGUgY29uc3RhbnRzIGZvciAxMDMgYW5kIDE2IHdpdGggZGVzY3JpcHRpdmUgbmFtZXMgdG8gZXhwbGFpbiB0aGVpciBzaWduaWZpY2FuY2UuIiwiZXhwIjoxNzUwODM2NzY5fQ.Q23WH7_IRpOEsHVEuxiiQVCbqAZOhjFM6UbFPb3WTDA)
